feat: nytt forsøk for å få scroll-effekten på første siden til å fungere

lpbjorkeng opened the pull request.

Commit:
feat: nytt forsøk for å få scroll-effekten på første siden til å fungere

mortentangen commented:
Du må ha en sjekk på om window er tilgjengelig ettersom vi Gatsby belager seg på server side rendering
![image](https://user-images.githubusercontent.com/499835/140029321-e02d0999-fe12-4042-8b2b-f66e94897b05.png)

lpbjorkeng commented:
Gatsby er ikke glad i bruk av window. Bør lære meg å kjøre build før jeg prøver meg på en pull request. Fikser og lager ny pull request.

mortentangen commented on src/pages/index.js:
```diff
@@ -70,24 +70,52 @@ const IndexPage = () => {
+        `circle(calc(${1.5 * window.scrollY}px + min(45vh, 45vw)) at 85% 70%)` :
+        `circle(calc(${1 * window.scrollY}px + 20vh) at 85% 90%)`
+    }
+    setClipPath();
```
så initiell clipPath, deretter endres den ved scroll?

lpbjorkeng replied:
Korrekt. Så linje 81 setter initiell clipPath til et høvelig pent sirkel-utsnitt ved load og så oppdateres den ved scroll. Satt først initiell clipPath i stylingen (linje 211) som helgardering, men det ble litt for mye repetisjon når man skal ta hensyn til flere typer skjermer.

Commit:
Sjekker at window er defined så build ikke feiler.
Siden bør nå også fungere greit selv om window på en eller annen måte ikke er definert.

lpbjorkeng commented:
Ser den såvidt flasher rosa på load her. Mulig jeg burde sette clipPath i stylingen likevel.

mortentangen commented:
> Ser den såvidt flasher rosa på load her. Mulig jeg burde sette clipPath i stylingen likevel.

ja, ser slik ut. Ser også at jeg må scrolle et lite stykke til etter at sirkelen er ute av syne før siden beveger seg nedover. Ellers virker dette veldig bra!

lpbjorkeng commented:
Jepp. Det med at man må scrolle litt forbi er bittelitt med vilje. Vil heller at man skal måtte scrolle litt ekstra enn at man på noen skjermer (veldig avlange telefoner på høykant) ikke får ferdig animasjonen før man scroller ned siden.
Kommer en commit til.

Commit:
Liten fix for mobilsjerm og for å unngå rosa blink på load.
